duckduckgo/duckduckgo-privacy-extension
# Different approach to parsing/serving surrogate content

**Reviewer:** @jdorweiler 



## Description:

This PR sets up the extension to detect outbound requests to GA and return data redirects to noop surrogate functions in their place.

As of now we're only addressing a handful of necessary cases from that list, but we've structured this so that we may add more without needing to change any code here.

## Steps to test this PR:
1. npm run dev-firefox (or chrome)
2. reload extension
3. in the console, check that surrogateList exists and is properly structured data
4. nuke whitelist `whitelists.trackersWhitelist.isLoaded = false`
5. visit broken site (I've been using raspberrypi.org, follow steps to buy)
6. request to GA should be intercepted, and a redirect to a base64 data url should be returned.
7. most importantly, the tracker is blocked and the page still functions.

You can further verify that the tracker was blocked by checking the network tab like this:
![virtualbox_2018-01-12_18-11-23](https://user-images.githubusercontent.com/4481594/34899617-0e3dd5c6-f7c8-11e7-933f-d377bb6f2bab.png)




## Automated tests:

###### Reviewer Checklist:


###### PR Author Checklist:
